src/components/Map.js

```diff
@@ -85,12 +86,11 @@ class Map extends Component {
 
       // For each marker created, add a listener that checks for clicks
       google.maps.event.addListener(marker, 'click', function (){ //eslint-disable-line
-        if (currentSelectedMarker) {
-          currentSelectedMarker.message.close();
-        } else {
-          context.props.changeCurrEvent(pin.eventIdx);
-        }
-        currentSelectedMarker = pin;
+        if (context.currentSelectedMarker) {
+          context.currentSelectedMarker.message.close();
+        } 
+        context.props.changeCurrEvent(pin.eventIdx);
+        context.currentSelectedMarker = pin;
```

Comment:
Suggest adding comments:

- above the if statement, add that the .message.close() function is on the pin object. I was a little confused when reviewing the code from top to bottom where this function was coming from. I guess it was confusing to have pin renamed current selected marker.